galaxy_ng/app/models/auth.py

```diff
@@ -23,7 +30,45 @@ def create_identity(self, scope, name):
         return super().create(name=self._make_name(scope, name))
 
     def get_or_create_identity(self, scope, name):
-        return super().get_or_create(name=self._make_name(scope, name))
+        group, _ = super().get_or_create(name=self._make_name(scope, name))
+
+        # First login for group/org, we need to create a AnsibleDistribution that points to
+        # the "golden" Repository
+        default_repo = self._create_repo(group, default_repo_name)
+        distro = self._create_distribution(group, default_repo)
+
+        log.debug('distro: %s', distro)
+        log.debug('group: %s', group)
+        log.debug('distro.repository: %s', distro.repository)
+
+        return group, _
+
+    @staticmethod
+    def _create_repo(group, repository_name):
+        # TODO: the lifetime of the default repo is long, so shouldn't need to look it
+        #       up on every auth attempt. ie, cache it.
+        # TODO: This may be a reasonable place to hook up a django signal
+        try:
+            default_repo = AnsibleRepository.objects.get(name=repository_name)
+        except AnsibleRepository.DoesNotExist as exc:
+            log.exception(exc)
+            raise
+
+        return default_repo
+
+    @staticmethod
+    def _create_distribution(group, repository):
+        distro_name = settings.GALAXY_API_SYNCLIST_NAME_FORMAT.format(
+            account_number=group.account_number())
+
+        try:
+            distro = AnsibleDistribution.objects.get(name=distro_name, base_path=distro_name)
+        except AnsibleDistribution.DoesNotExist as exc:
+            log.debug('distro_name: %s exc: %s', distro_name, exc)
```

Comment:
Probably makes sense to add operation name that is executed in addition to some local data.